Dockerfile

```diff
@@ -2,36 +2,33 @@ FROM node:10.13-alpine
 
 WORKDIR /app
 
+ENV PORT 3005
+EXPOSE 3005
+
 # Ensure COMMIT_HASH is present
 ARG COMMIT_HASH
 RUN test -n "$COMMIT_HASH"
+RUN echo $COMMIT_HASH > COMMIT_HASH.txt
```

Comment:
Since this value of `$COMMIT_HASH` is likely to change often, I think we should move it down so it doesn't interfere with the caching of other layers. The downside of moving these lines further down would be that you spend time building the docker image only to find out later that there's a required argument, but I think that's a reasonable trade off to ensure that we best utilize the docker layer caching mechanism.

Another tip is to collapse any neighboring `RUN` instructions to limit the total amount of docker layers defined.

For example:

```dockerfile
RUN test -n "$COMMIT_HASH" && \
    echo $COMMIT_HASH > COMMIT_HASH.txt
```